### PR TITLE
feat(windows): add smoke test subcommand

### PR DESCRIPTION
**Pull request by ReactorScram**
Part of #3534
This PR creates the subcommand, which you can run locally, but it doesn't run it in `ci.yml` yet. That's in #3542

### CODE REVIEW

File: rust/windows-client/src-tauri/src/client/gui.rs [hunk 1]
```diff
@@ -265,6 +275,54 @@ pub(crate) fn run(cli: client::Cli) -> Result<(), Error> {
...
+/// Runs a smoke test and then asks Controller to exit gracefully
+///
+/// You can purposely fail this test by deleting the exported zip file during
+/// the 10-second sleep.
+async fn smoke_test(ctlr_tx: CtlrTx) -> Result<()> {
+    let delay = 10;
+    tracing::info!("Will quit on purpose in {delay} seconds as part of the smoke test.");
+    let quit_time = tokio::time::Instant::now() + Duration::from_secs(delay);
+
+    // Test log exporting
```
Comment (jamilbk) [inline rust/windows-client/src-tauri/src/client/gui.rs:291]: 🤩

**Review by jamilbk — COMMENTED** [review]
LGTM, but deferring to @conectado for approval on the Rust code.

It would be nice to find some way to get the clients tested in GH actions, even minimally somehow, as opposed to trying to run automated tests on the janky testbed. I don't think there is, but I'm still holding out hope

File: rust/windows-client/src-tauri/src/client.rs [hunk 2]
```diff
@@ -89,7 +89,7 @@ pub(crate) fn run() -> Result<()> {
         Some(Cmd::CrashHandlerServer { socket_path }) => crash_handling::server(socket_path),
         Some(Cmd::Debug { command }) => debug_commands::run(command),
         // If we already tried to elevate ourselves, don't try again
-        Some(Cmd::Elevated) => run_gui(cli),
+        Some(Cmd::Elevated | Cmd::SmokeTest) => run_gui(cli),
```
Comment (conectado) [inline rust/windows-client/src-tauri/src/client.rs]: I'd rather we inline the function here so we can split off this pattern matching, I think it's clearer to follow the flow of this function.

File: rust/windows-client/src-tauri/src/client.rs [hunk 3]
```diff
@@ -102,26 +102,33 @@ pub(crate) fn run() -> Result<()> {
...
-            error => format!("{}", error),
-        };
-
-        native_dialog::MessageDialog::new()
-            .set_title("Firezone Error")
-            .set_text(&error_msg)
-            .set_type(native_dialog::MessageType::Error)
-            .show_alert()?;
+
+        if let Some(Cmd::SmokeTest) = &cli.command {
```
Comment (conectado) [inline rust/windows-client/src-tauri/src/client.rs]: We wouldn't need this branch if we follow the structure I laid out in the previous comment, however, if we keep this we can just:
```rs
if let Some(Cmd::SmokeTest) = &cli.command { return Ok(()); }
```

File: rust/windows-client/src-tauri/src/client.rs [hunk 4]
```diff
@@ -102,26 +102,33 @@ pub(crate) fn run() -> Result<()> {
...
+            // In smoke-test mode, don't show the dialog, since it might be running
+            // unattended and the dialog would hang forever
+        } else {
+            let error_msg = match &error {
+                gui::Error::WebViewNotInstalled => "Firezone cannot start because WebView2 is not installed. Follow the instructions at <https://www.firezone.dev/kb/user-guides/windows-client>.".to_string(),
+                error => format!("{}", error),
```
Comment (conectado) [inline rust/windows-client/src-tauri/src/client.rs]:
```suggestion
                error => error.to_string(),
```

File: rust/windows-client/src-tauri/src/client.rs (hunk 3 above)
Comment (ReactorScram) [reply rust/windows-client/src-tauri/src/client.rs]: I still want it to return the error so CI will see it failed, but yeah I'm removing the branch anyway

**Review by conectado — APPROVED** [review]

**Pull request merged**